src/main/java/pl/szczesniak/dominik/whattowatch/users/domain/UserService.java

```diff
@@ -19,9 +21,13 @@ public boolean exists(final UserId userId) {
 		return repository.exists(userId);
 	}
 
-
-	public UserId login(String username) {
-		return repository.findBy(username).get().getUserId();
+	public UserId login(final Username username, final UserPassword userPassword) {
+		final User user = repository.findBy(username.getValue()).orElseThrow(() -> new WrongUsernameOrPasswordException("User with username: " + username.getValue() + " not found"));
```

Comment:
spróbuj to zrobić używając metody filter na optionalu. Będziesz się mógł pozbyć tego ifa niżej